framework/web_app/src/stores/ProductStore.ts

```diff
@@ -0,0 +1,163 @@
+import { defineStore } from 'pinia';
+import { Loading } from 'quasar';
+import { IOfferSortByOption, OfferSortByOptions } from 'src/helpers/OfferSortByOptions';
+import { isOfferFavourited } from 'src/helpers/ScrapedProductOfferLogic';
+import { Merchant } from 'src/models/Merchant';
+import type { Product } from 'src/models/Product';
+import type { ScrapedProductOffer } from 'src/models/ScrapedProductOffer';
+import MerchantApiService from 'src/services/api/MerchantApiService';
+import ProductApiService, { SearchByTermQuery } from 'src/services/api/ProductApiService';
+import { computed, reactive, ref } from 'vue';
+
+const merchantApiService = new MerchantApiService();
+const productApiService = new ProductApiService();
+
+// TODO:
+// New Features:
+// - Create a relevance algorithm & set as default sortBy
+// - Add ability to un-favourite a product
+// - Add query functionality on get products.
+    //In the .Then for addOfferToFavouritesAsync(),
+    //add the new product to state using a query specifically for that product rather than getting all products.
+// - Add quasar notify/toast OR other feedback for when a product is favourited
+
+// TODO:
+// Issues:
+// - Scraped offers return merchant stock code as a number BUT get products returns it as a string
```

Comment:
probably want a string, no? i think stockcode could be anything

Reply:
is a string now